cpp/src/barretenberg/honk/sumcheck/sumcheck.hpp

```diff
@@ -12,17 +14,22 @@
 #include <string>
 #include <vector>
 #include "barretenberg/honk/proof_system/prover.hpp"
+#include "barretenberg/honk/sumcheck/sumcheck_output.hpp"
+#include <optional>
+
 namespace honk::sumcheck {
+
 template <typename FF, class Transcript, template <class> class... Relations> class Sumcheck {
```

Comment:
I think we can remove the `Transcript` template parameter now.

Reply:
Definitely. I got carried away with the "minimal changes" mantra..

Reply:
`Transcript` currently specializes to `ProverTranscript` and `VerifierTranscript`. If someone wants to make argument for splitting the whole library in to prover and verifier versions of everything, I'll listen, but I'm not in favor without a compelling argument. I'm really not even in favor of this for the `Transcript`, where I think the mental model of a single transcript between two parties reflects better the mental model of zk protocols, has essentially no implementation cost, and agrees with what is done in `plonk`.